File: ESC190/midterm/2024/ll.c
#include <stdio.h>
#include <stdlib.h>

typedef struct node{
    int data;
    struct node *next;
} node;

typedef struct LL{
    node *head;
} LL;

int len_ll(LL *list)
{
    int i = 1;
    if (list->head == NULL)
    {
        return 0;
    }
    node *elem = list->head;
    while (elem->next != NULL)
    {
        i++;
        elem = elem->next;
    }
    return i++;
}

int main()
{
    LL *list = (LL *)malloc(sizeof(LL));
    list->head = (node *)malloc(sizeof(node));
    list->head->data = 1;
    list->head->next = NULL;
    list->head->next->data = 2;
    list->head->next->next = (node *)malloc(sizeof(node));
    // list->head->next->next->data = 3;
    // list->head->next->next->next = NULL;
    printf("%d\n", len_ll(list));
    return 0;
}

File: ESC190/labs/lab11/ll.py
class Node():
    def __init__(self, value):
        self.value = value
        self.next = None

class LL():
    def __init__(self):
        self.head = None
        self.tail = None
        
    def append(self, value):
        if self.head == None:
            self.head = Node(value)
            self.tail = self.head
        else:
            self.tail.next = Node(value)
            self.tail = self.tail.next